packages/ethers-5/src/client.ts

```diff
@@ -0,0 +1,788 @@
+    }
+
+    const networkControllerClient: NetworkControllerClient = {
+      switchCaipNetwork: async caipNetwork => {
+        const chainId = caipNetworkIdToNumber(caipNetwork?.id)
+        if (chainId) {
+          await this.switchNetwork(chainId)
+        }
+      },
+
+      getApprovedCaipNetworksData(): Promise<{
+        approvedCaipNetworkIds: `${string}:${string}`[] | undefined
+        supportsAllNetworks: boolean
+      }> {
+        return new Promise(resolve => {
+          const walletChoice = localStorage.getItem(WALLET_ID)
+
+          if (walletChoice?.includes(WALLET_CONNECT_CONNECTOR_ID)) {
+            const connector = ethersConfig.walletConnect
+            if (!connector) {
+              throw new Error(
+                'networkControllerClient:getApprovedCaipNetworks - connector is undefined'
+              )
+            }
+            const ns = (ethersConfig.walletConnect?.provider as EthereumProvider).signer?.session
+              ?.namespaces
+            const nsMethods = ns?.[NAMESPACE]?.methods
+            const nsChains = ns?.[NAMESPACE]?.chains
+
+            const result = {
+              supportsAllNetworks: nsMethods?.includes(ADD_CHAIN_METHOD) ?? false,
+              approvedCaipNetworkIds: nsChains as CaipNetworkId[] | undefined
+            }
+
+            resolve(result)
+          } else {
+            const result = {
+              approvedCaipNetworkIds: undefined,
+              supportsAllNetworks: true
+            }
+
+            resolve(result)
+          }
+        })
+      }
+    }
+
+    const connectionControllerClient: ConnectionControllerClient = {
+      connectWalletConnect: async onUri => {
+        const connector = ethersConfig.walletConnect
+        if (!connector) {
+          throw new Error('connectionControllerClient:getWalletConnectUri - connector is undefined')
+        }
+
+        const walletConnectProvider = connector.provider as EthereumProvider
+
+        walletConnectProvider.on('display_uri', (uri: string) => {
+          onUri(uri)
+        })
+
+        await walletConnectProvider.connect().then(() => {
+          this.setWalletConnectProvider(ethersConfig)
+        })
+      },
+
+      connectExternal: async ({ id, info, provider }) => {
+        if (id === INJECTED_CONNECTOR_ID) {
+          const injectedProvider = ethersConfig.injected
+          if (!injectedProvider) {
+            throw new Error('connectionControllerClient:connectInjected - connector is undefined')
+          }
+          await injectedProvider.send('eth_requestAccounts', []).then(() => {
+            this.setInjectedProvider(ethersConfig)
+          })
+        } else if (id === EIP6963_CONNECTOR_ID && info && provider) {
+          const EIP6963Provider = provider as ethers.providers.Web3Provider
+          const EIP6963Info = info as Info
+
+          await EIP6963Provider.send('eth_requestAccounts', []).then(() => {
```

Comment:
Inconsistent casings in provider name, I think injectedProvider / coinbaseProvider should be `InjectedProvider` and `CoinbaseProvider`, this will make them look more similar to wagmi's connectors as well.